trade-finance-manager-ui/server/controllers/case/cancellation/check-details.controller.ts

```diff
@@ -0,0 +1,82 @@
+import { Response } from 'express';
+import { CustomExpressRequest } from '@ukef/dtfs2-common';
+import { format } from 'date-fns';
+import { PRIMARY_NAVIGATION_KEYS } from '../../../constants';
+import { asUserSession } from '../../../helpers/express-session';
+import { canSubmissionTypeBeCancelled } from '../../helpers/deal-cancellation-enabled.helper';
+import api from '../../../api';
+import { CheckDetailsViewModel } from '../../../types/view-models';
+
+export type GetDealCancellationDetailsRequest = CustomExpressRequest<{ params: { _id: string } }>;
+export type PostDealCancellationDetailsRequest = CustomExpressRequest<{ params: { _id: string } }>;
+
+/**
+ * controller to get deal cancellation details
+ *
+ * @param req - The express request
+ * @param res - The express response
+ */
+export const getDealCancellationDetails = async (req: GetDealCancellationDetailsRequest, res: Response) => {
+  const { _id } = req.params;
+  const { user, userToken } = asUserSession(req.session);
+
+  try {
+    const deal = await api.getDeal(_id, userToken);
+
+    if (!deal || 'status' in deal) {
+      return res.redirect('/not-found');
+    }
+
+    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
+      return res.redirect(`/case/${_id}/deal`);
+    }
+
+    const { reason, bankRequestDate, effectiveFrom } = await api.getDealCancellation(_id, userToken);
+
+    const bankRequestDateFormatted = bankRequestDate ? format(new Date(bankRequestDate), 'd MMMM yyyy') : undefined;
```

Comment:
```suggestion
    const bankRequestDateFormatted = bankRequestDate && format(new Date(bankRequestDate), 'd MMMM yyyy');
```

Reply:
Due to using Typescript these two implementations are subtly different: 

**Current implementation:**
![image](https://github.com/user-attachments/assets/a7d40e38-a402-42cd-a521-f47db7e7daa8)
`bankRequestDateFormatted` returns either a `string` or `undefined`



**Suggested implementation:**
![image](https://github.com/user-attachments/assets/c1094059-6eb2-489c-b423-c72f8251d511)
`bankRequestDateFormatted` returns either a `string`, `undefined` OR `0`
This is due to possibly surprising behaviour / took me a while to think about! But bankRequestDate is received as either a number or undefined, and obviously when it's a number we expect it to be a Unix time stamp, but Typescript doesn't know this so it think the number _could be 0_. So if we use the `&&` operator, if it was hypothetically `0` then it would return `0` as the LHS of the operand which is why it considers this a possible outcome of the statement. 

Due to it being a typescript file, this causes problems elsewhere in the file if it thinks it could be 0 (even though we know it won't ever be). So I'd rather keep it as is to make it clear that we're passing in `undefined` or a string. 

What do you reckon?